ddtrace/contrib/botocore/patch.py

```diff
@@ -412,10 +456,65 @@ def patched_api_call(original_func, instance, args, kwargs):
         span.set_tag(ANALYTICS_SAMPLE_RATE_KEY, config.botocore.get_analytics_sample_rate())
 
         try:
-            result = original_func(*args, **kwargs)
-            _set_response_metadata_tags(span, result)
+            if endpoint_name == "sqs" and operation == "ReceiveMessage" and config._data_streams_enabled:
+                queue_url = params['QueueUrl']
+                queue_name = queue_url[queue_url.rfind("/") + 1:]
+
+                #if needed, modify args to fetch the _datadog data
+
+                query_status = None
+                if 'MessageAttributeNames' not in args[1]:
+                    query_status = "NoMessageAttributeNames"
+                    args = (args[0], {**args[1], 'MessageAttributeNames': ['_datadog']})
+                elif '_datadog' not in args[1]['MessageAttributeNames']:
+                    query_status = "NoDatadog"
+                    args = (args[0], {**args[1], 'MessageAttributeNames': args[1]['MessageAttributeNames'] + ['_datadog']})
+
+                result = original_func(*args, **kwargs)
+                _set_response_metadata_tags(span, result)
+
+                for message in result['Messages']:
+                    try:
+                        #TODO: potentially deal with the case where you
+                        # need to remove _datadog info from result
+                        # even when dd-pathway-ctx isn't there
+                        pathway = json.loads(
+                            message['MessageAttributes']['_datadog']['StringValue']
+                        )['dd-pathway-ctx']
+
+                        ctx = pin.tracer.data_streams_processor.decode_pathway_b64(pathway)
+                        ctx.set_checkpoint(
+                            ["direction:in", "topic:" + queue_name, "type:sqs"]
+                        )
+
+                    except:
+                        return result
+
+                if query_status == "NoMessageAttributeNames":
+                    if 'Messages' in result:
+                        messages = result['Messages']
+                        for message in messages:
+                            if 'MessageAttributes' in message:
+                                del message['MessageAttributes']
+
+                elif query_status == "NoDatadog":
```

Comment:
make this another method with a descriptive name.  The method name should be a good summary of what and why we're doing this